torch/nn/utils/spectral_norm.py

```diff
@@ -0,0 +1,118 @@
+"""
+Spectral Normalization from https://arxiv.org/abs/1802.05957
+"""
+import torch
+from torch.nn.functional import normalize
+from torch.nn.parameter import Parameter
+
+
+class SpectralNorm(object):
+
+    def __init__(self, name='weight', n_power_iterations=1, eps=1e-12):
+        self.name = name
+        self.n_power_iterations = n_power_iterations
+        self.eps = eps
+
+    def compute_weight(self, module):
+        weight = module._parameters[self.name + '_org']
+        u = module._buffers[self.name + '_u']
+        height = weight.size(0)
+        weight_mat = weight.view(height, -1)
+        for _ in range(self.n_power_iterations):
+            # Spectral norm of weight equals to `u^T W v`, where `u` and `v`
+            # are the first left and right singular vectors.
+            # This power iteration produces approximations of `u` and `v`.
+            v = normalize(torch.matmul(weight_mat.t(), u), dim=0, eps=self.eps)
+            u = normalize(torch.matmul(weight_mat, v), dim=0, eps=self.eps)
+
+        sigma = torch.dot(u, torch.matmul(weight_mat, v))
+        weight.data /= sigma
```

Comment:
Why is it made non differentiable?

Reply:
A couple bugs of this initial implementation is fixed before release, including making it differentials.